app/controllers/claims_controller.rb

```diff
@@ -14,46 +14,19 @@ def new
     }
   end
 
-  # FIXME: extract into an interactor
   def create
-    # TODO: Need to handle race conditions to prevent creating multiple matches for same contribution.
-    contribution = Listing.find(params[:contribution_id])
-    ActiveRecord::Base.transaction do
-      if current_person.blank?
-        Person.create_from_peer_to_peer_params!(current_user, name: claim_params[:peer_alias],
-                                                preferred_contact_method_id: claim_params[:preferred_contact_method_id],
-                                                contact_info: claim_params[:preferred_contact_info])
-      elsif current_person.present? && current_person.email.blank?
-        current_person.update!(email: claim_params[:preferred_contact_info])
-      end
-      Match.create_match_for_contribution!(contribution, current_user)
-      contribution.matched!
-    end
-    notify_peer_and_log_communication!(contribution)
+    ClaimContribution.run!(
+      contribution: params[:contribution_id],
+      claim_params: claim_params.to_h,
+      current_user: current_user
+    )
```

Comment:
I'd consider flattening the `claim_params` argument in the interactor into the three separate params, so that this code can be:
```suggestion
    ClaimContribution.run!(params.merge(
      contribution: params[:contribution_id],
      current_user: current_user
    ))
```